app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt

```diff
@@ -155,35 +188,97 @@ class ConversationViewModel(
     )
 
     init {
-        viewModelScope.launch(Dispatchers.IO) {
+        viewModelScope.launch(Dispatchers.Default) {
             repository.recipientUpdateFlow(threadId)
                 .collect { recipient ->
-                    if (recipient == null && _uiState.value.conversationExists) {
-                        _uiState.update { it.copy(conversationExists = false) }
+                    _uiState.update {
+                        it.copy(
+                            shouldExit = recipient == null,
+                            showInput = shouldShowInput(recipient),
+                            enableInputMediaControls = shouldEnableInputMediaControls(recipient),
+                            messageRequestState = buildMessageRequestState(recipient),
+                        )
                     }
                 }
         }
+    }
 
-        // listen to community write access updates from this point
-        communityWriteAccessJob?.cancel()
-        communityWriteAccessJob = viewModelScope.launch {
-            OpenGroupManager.getCommunitiesWriteAccessFlow()
-                .map {
-                    if(openGroup?.groupId != null)
-                        it[openGroup?.groupId]
-                    else null
-                }
-                .filterNotNull()
-                .collect{
-                    // update our community object
-                    _openGroup.updateTo(openGroup?.copy(canWrite = it))
-                    // when we get an update on the write access of a community
-                    // we need to update the input text accordingly
-                    _uiState.update { state ->
-                        state.copy(hideInputBar = shouldHideInputBar())
-                    }
+    /**
+     * Determines if the input media controls should be enabled.
+     *
+     * Normally we will show the input media controls, only in these situations we hide them:
+     *  1. First time we send message to a person.
+     *     Since we haven't been approved by them, we can't send them any media, only text
+     */
+    private fun shouldEnableInputMediaControls(recipient: Recipient?): Boolean {
+        if (recipient != null &&
+            (recipient.is1on1 && !recipient.isLocalNumber) &&
+            !recipient.hasApprovedMe()) {
+            return false
+        }
+
+        return true
+    }
+
+    /**
+     * Determines if the input bar should be shown.
+     *
+     * For these situations we hide the input bar:
+     *  1. The user has been kicked from a group(v2), OR
+     *  2. The legacy group is inactive, OR
+     *  3. The community chat is read only
```

Comment:
This third property is dynamic and can change while the conversation is open, hence the `getCommunitiesWriteAccessFlow` code above which was deleted. Is this handled somewhere else now?

Reply:
Just like my previous comment, I'd expect recipient update flow to go through for a community info change. If not, we should make sure the change is flowing through the recipient change instead

Reply:
This is no longer reactive as tested with the SOGS